scripts/simulate_live_data.py

```diff
@@ -25,10 +27,12 @@ def copy_dataset(source_dir, dest_dir, rate, slow_copy_mode):
         source_item = os.path.join(source_dir, item)
         dest_item = os.path.join(dest_dir, item)
         if os.path.isfile(source_item):
-            if not slow_copy_mode:
-                shutil.copy(source_item, dest_item)
-            else:
+            if slow_copy_mode:
                 slow_copy(source_item, dest_item)
+            elif faulty_copy_mode:
+                faulty_copy(source_item, dest_item)
+            else:
+                shutil.copy(source_item, dest_item)
```

Comment:
I think you may need to modify line 39 to:
```python
copy_dataset(source_item, dest_item, rate, slow_copy_mode, faulty_copy_mode)
```
to avoid:
```bash
  0%|                                                                                                   | 0/3 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/vco13151/mantidimaging/scripts/simulate_live_data.py", line 101, in <module>
    main()
  File "/home/vco13151/mantidimaging/scripts/simulate_live_data.py", line 95, in main
    copy_dataset(args.source_dir, args.dest_dir, args.rate, args.slow, args.faulty)
  File "/home/vco13151/mantidimaging/scripts/simulate_live_data.py", line 39, in copy_dataset
    copy_dataset(source_item, dest_item, rate, slow_copy_mode)
TypeError: copy_dataset() missing 1 required positional argument: 'faulty_copy_mode'
```
when running from the cli with: `python simulate_live_data.py -s ../mantidimaging-data-small/ -d ../live_dir_test/ --faulty`

Reply:
ah, is then when doing subfolders?

Reply:
when using any source dir for me. are you able to replicate?

Reply:
I think it can only be hit if you have a subdir.

Ive added another commit with type annotations that would have caught the omission of the argument.

Reply:
Thanks, and yes, seemed to be when hitting a subdir. Executes successfully now.